src/lib/LLVMExec.hs

```diff
@@ -153,6 +156,13 @@ standardCompilationPipeline logger exports tm m = do
 
 -- === object file export ===
 
+exportObjectFileVal :: [ObjectFileName n] -> L.Module -> String -> IO (ObjectFile n)
+exportObjectFileVal deps m fname = do
+  withSystemTempFile "objfile.o" \path h -> do
+    exportObjectFile path [(m, [fname])]
+    contents <- B.hGetContents h
+    return $ ObjectFile contents [fname] deps
```

Comment:
This is very roundabout. You could just do `Mod.moduleObject` instead of `Mod.writeObjectToFile` and avoid the unnecessary filesystem indirection.